src/model/ValidatorFactory.js

```diff
@@ -9,69 +9,139 @@ import FormUtils from "../util/FormUtils";
 
 export default class ValidatorFactory {
   static createValidator(question, intl) {
-    if (question[Constants.REQUIRES_ANSWER]) {
-      if (FormUtils.isCheckbox(question)) {
-        //TODO revise
-        return ValidatorFactory._generateRequiresAnswerCheckBoxValidator(
-          question,
-          intl
-        );
+    const validators = [
```

Comment:
This order to me is still confusing me.  You said that it does not matter where i put the patternValidator because of the specific algorithm that you ,mplemented. But the order should be meaningful regardless of the specifics of the implementation, and so the only order that makes sense to me is (with a hope that i understand what each validator does :):
```
// checking for invalid values (if there is a value, it must be valid)
this._patternValidator,
// checking for values required to save
this._checkboxValidator,
this._requiredValidator,
// checking for values required to complete
this._completenessValidator,
```

The exact opposite order is of course, meaningful to me as well.